pkg/apis/pipeline/v1/artifact_types.go

```diff
@@ -19,3 +19,8 @@ type ArtifactValue struct {
 // TaskRunStepArtifact represents an artifact produced or used by a step within a task run.
 // It directly uses the Artifact type for its structure.
 type TaskRunStepArtifact = Artifact
+
+type Artifacts struct {
+	Inputs  []Artifact `json:"inputs"`
+	Outputs []Artifact `json:"outputs"`
+}
```

Comment:
```suggestion
type Artifacts struct {
 	Inputs  []Artifact `json:"inputs,omitempty"`
 	Outputs []Artifact `json:"outputs,omitempty"`
 }
```

Reply:
This way, it won't be enforced. 

If a step only produces an artifact, they will likely only provide `onputs` not `inputs`. In that case, Inputs should not be listed at all.

Reply:
Same for `v1beta1` as well.

Reply:
thanks, I'll update